src/postgres_client/postgres_client_account_index.rs

```diff
@@ -0,0 +1,377 @@
+        &mut self,
+        token_id: &Pubkey,
+        account: &DbAccountInfo,
+    ) {
+        if account.owner() == token_id.to_bytes() {
+            if let Some(mint_key) = G::unpack_account_mint(account.data()) {
+                let mint_key = mint_key.to_bytes().to_vec();
+                let pubkey = account.pubkey();
+                self.pending_token_mint_index.push(TokenSecondaryIndex {
+                    owner: mint_key,
+                    account_key: pubkey.to_vec(),
+                    slot: account.slot,
+                })
+            }
+        }
+    }
+
+    /// Queue bulk insert secondary indexes: token owner and token mint indexes.
+    pub fn queue_secondary_indexes(&mut self, account: &DbAccountInfo) {
+        if self.index_token_owner {
+            self.queue_token_owner_index_generic::<inline_spl_token::Account>(
+                &inline_spl_token::id(),
+                account,
+            );
+            self.queue_token_owner_index_generic::<inline_spl_token_2022::Account>(
+                &inline_spl_token_2022::id(),
+                account,
+            );
+        }
+
+        if self.index_token_mint {
+            self.queue_token_mint_index_generic::<inline_spl_token::Account>(
+                &inline_spl_token::id(),
+                account,
+            );
+            self.queue_token_mint_index_generic::<inline_spl_token_2022::Account>(
+                &inline_spl_token_2022::id(),
+                account,
+            );
+        }
+    }
+
+    /// Generic function to update a single token owner index.
+    fn update_token_owner_index_generic<G: GenericTokenAccount>(
+        client: &mut Client,
+        statement: &Statement,
+        token_id: &Pubkey,
+        account: &DbAccountInfo,
+    ) -> Result<(), AccountsDbPluginError> {
+        if account.owner() == token_id.to_bytes() {
+            if let Some(owner_key) = G::unpack_account_owner(account.data()) {
+                let owner_key = owner_key.to_bytes().to_vec();
+                let pubkey = account.pubkey();
+                let slot = account.slot;
+                let result = client.execute(statement, &[&owner_key, &pubkey, &slot]);
+                if let Err(err) = result {
+                    let msg = format!(
+                        "Failed to update the token owner index to the PostgreSQL database. Error: {:?}",
+                        err
+                    );
+                    error!("{}", msg);
+                    return Err(AccountsDbPluginError::AccountsUpdateError { msg });
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Generic function to update a single token mint index.
+    fn update_token_mint_index_generic<G: GenericTokenAccount>(
+        client: &mut Client,
+        statement: &Statement,
+        token_id: &Pubkey,
+        account: &DbAccountInfo,
+    ) -> Result<(), AccountsDbPluginError> {
+        if account.owner() == token_id.to_bytes() {
```

Comment:
nittiest of nits: I imagine this is a hot path, so you can potentially speed this up by replacing `to_bytes()` with `as_ref()` everywhere

Reply:
Great! Thanks